### PR TITLE
bcc/tools: Add biopattern.py to identify disk access patterns

### PR DESCRIPTION
**Pull request by xingfeng2510**
Add biopattern.py (Based on libbpf-tools/biopattern) to identify random/sequential disk access patterns.

Example:
```
# ./biopattern.py -d vdb 1 10
TIME      DISK     %RND  %SEQ    COUNT     KBYTES
22:12:57  vdb         0    99      193        772
22:12:58  vdb         0   100     1119       4476
22:12:59  vdb         0   100     1126       4504
22:13:00  vdb         0   100     1009       4036
22:13:01  vdb         0   100      958       3832
22:13:02  vdb         0    99      957       3856
22:13:03  vdb         0   100     1130       4520
22:13:04  vdb         0   100     1051       4204
22:13:05  vdb         0   100     1158       4632
[...]
```

@yonghong-song @davemarchevsky please help review this pr, thanks.

### CODE REVIEW

File: tools/biopattern.py [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+            dev_num = major << dev_minor_bits | minor
+            bpf_text = bpf_text.replace('DISK_FILTER',
+                                        'if (dev != %s) { return 0; }' % dev_num)
+            break
+
+    if found == 0:
+        print("no such disk '%s'" % args.disk)
+        exit(1)
+else:
+    bpf_text = bpf_text.replace('DISK_FILTER', '')
```
Comment (chenhengqi) [inline tools/biopattern.py:103]: Use stat `/dev/xxx`.

File: tools/biopattern.py [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+        return 0;
+    }
+
+    if (counterp->last_sector) {
+        if (counterp->last_sector == sector) {
+            __sync_fetch_and_add(&counterp->sequential, 1);
+        } else {
+            __sync_fetch_and_add(&counterp->random, 1);
+        }
+        __sync_fetch_and_add(&counterp->bytes, nr_sector * 512);
```
Comment (chenhengqi) [inline tools/biopattern.py:70]: I think this line should move out of if block.

File: tools/biopattern.py [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+parser.add_argument("interval", nargs="?", default=99999999,
+    help="Output interval in seconds")
+parser.add_argument("count", nargs="?", default=99999999,
+    help="Number of outputs")
+args = parser.parse_args()
+countdown = int(args.count)
+
+bpf_text="""
+#include <uapi/linux/ptrace.h>
+#include <linux/blkdev.h>
```
Comment (chenhengqi) [inline tools/biopattern.py:38]: These headers are unused.

Comment (xingfeng2510) [reply tools/biopattern.py]: done

File: tools/biopattern.py (hunk 1 above)
Comment (xingfeng2510) [reply tools/biopattern.py]: done, thanks.

File: tools/biopattern.py (hunk 2 above)
Comment (xingfeng2510) [reply tools/biopattern.py:69]: if counterp->last_sector equals 0, means counterp not found and inited。First time you can not determine whether this I/O is seq or random，so here add seq or random at next time（counterp->last_sector > 0）.

File: tools/biopattern.py [hunk 4]
```diff
@@ -0,0 +1,140 @@
+#!/usr/bin/python
+# @lint-avoid-python-3-compatibility-imports
+#
+# biopattern - Identify random/sequential disk access patterns.
+#              For Linux, uses BCC, eBPF.
+#
+# Copyright (c) 2022 rockyxing.
```
Comment (yonghong-song) [inline tools/biopattern.py:7]: Could you proper spell name here, "Feng Xing" or "Rocky Xing" or anything your prefer? You can check biolatency.py for an example.

File: tools/biopattern.py [hunk 5]
```diff
@@ -0,0 +1,140 @@
...
+# Licensed under the Apache License, Version 2.0 (the "License")
+#
+# 21-Feb-2022   rockyxing   Created this.
```
Comment (yonghong-song) [inline tools/biopattern.py:10]: The same here, a proper name is needed.

**Review by yonghong-song — COMMENTED** [review]
I checked biopattern.py vs. libbpf-tools version biopattern, the implementation is similar so we should be fine. Could you add a man page for this new tool? Please also add the tool to README.md.

**Comment by netedwardwu** [comment]
Verfied on ARM64 5.10 kernel.

And some results from pep8 coding style checking.
https://github.com/iovisor/bcc/blob/master/CONTRIBUTING-SCRIPTS.md
```
pep8 --show-source --ignore=E123,E125,E126,E127,E128,E302 biopattern.py
```
```
biopattern.py:36:9: E225 missing whitespace around operator
bpf_text="""
        ^
biopattern.py:72:4: E111 indentation is not a multiple of four
   return (major << dev_minor_bits) | minor
   ^
biopattern.py:88:34: E712 comparison to False should be 'if cond is False:' or 'if not cond:'
    if os.path.exists(disk_path) == False:
                                 ^
biopattern.py:96:80: E501 line too long (85 > 79 characters)
                                'if (dev != %s) { return 0; }' % mkdev(major, minor))
                                                                               ^
biopattern.py:105:37: W291 trailing whitespace
print("%-9s %-7s %5s %5s %8s %10s" %
                                    ^
biopattern.py:113:1: W293 blank line contains whitespace

^

```

File: tools/biopattern.py (hunk 4 above)
Comment (xingfeng2510) [reply tools/biopattern.py]: Done

File: tools/biopattern.py (hunk 5 above)
Comment (xingfeng2510) [reply tools/biopattern.py]: Done

**Comment by xingfeng2510** [comment]
> I checked biopattern.py vs. libbpf-tools version biopattern, the implementation is similar so we should be fine. Could you add a man page for this new tool? Please also add the tool to README.md.

Done. Thank you for your review.

File: tools/biopattern.py (hunk 5 above)
Comment (xingfeng2510) [reply tools/biopattern.py]: Done. Thank you for your review.

**Pull request merged**